File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/IndexOfWord/IndexOfWord_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.easy.IndexOfWord.IndexOfWord import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        sentence = 'Hello every world'
        word = 'every'

        # when
        result = solution(sentence, word)

        # then
        self.assertEqual(1, result)

    def test_2(self):
        # given
        sentence = 'i love eating burger'
        word = 'i'

        # when
        result = solution(sentence, word)

        # then
        self.assertEqual(0, result)

    def test_3(self):
        # given
        sentence = 'My name is Dave'
        word = 'Dave'

        # when
        result = solution(sentence, word)

        # then
        self.assertEqual(3, result)

    def test_4(self):
        # given
        sentence = 'Ut congue ante nec libero malesuada accumsan'
        word = 'acc'

        # when
        result = solution(sentence, word)

        # then
        self.assertEqual(-1, result)

    def test_5(self):
        # given
        sentence = ''
        word = 'every'

        # when
        result = solution(sentence, word)

        # then
        self.assertEqual(-1, result)

    def test_6(self):
        # given
        sentence = ''
        word = ''

        # when
        result = solution(sentence, word)

        # then
        self.assertEqual(-1, result)


if __name__ == '__main__':
    unittest.main()

File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/IndexOfWord/IndexOfWord_test.js
import assert from 'assert'
import solution from './IndexOfWord.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const sentence = 'Hello every world'
    const word = 'every'

    // when
    const result = solution(sentence, word)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const sentence = 'i love eating burger'
    const word = 'i'

    // when
    const result = solution(sentence, word)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const sentence = 'My name is Dave'
    const word = 'Dave'

    // when
    const result = solution(sentence, word)

    // then
    assert.deepStrictEqual(3, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const sentence = 'Ut congue ante nec libero malesuada accumsan'
    const word = 'acc'

    // when
    const result = solution(sentence, word)

    // then
    assert.deepStrictEqual(-1, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const sentence = ''
    const word = 'every'

    // when
    const result = solution(sentence, word)

    // then
    assert.deepStrictEqual(-1, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const sentence = ''
    const word = ''

    // when
    const result = solution(sentence, word)

    // then
    assert.deepStrictEqual(-1, result)
    done();
  });
})
